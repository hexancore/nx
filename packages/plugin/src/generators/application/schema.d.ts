import type { ApplicationType } from "./ApplicationType";

export interface ApplicationGeneratorSchema {
  directory: string;
  type: ApplicationType | 'backend' | 'frontend';
}