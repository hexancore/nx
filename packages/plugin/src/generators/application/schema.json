{
  "$schema": "https://json-schema.org/draft/2020-12/output/schema",
  "$id": "Application",
  "title": "",
  "type": "object",
  "properties": {
    "directory": {
      "type": "string",
      "description": "A directory where the application is placed in `apps/`. Project name pattern: 'acme/api' -> 'app-acme-api'",
      "x-prompt": "Application project directory in `apps/`",
      "x-priority": "important",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "type": {
      "description": "The application type.",
      "type": "string",
      "enum": ["backend", "frontend"],
      "x-prompt": "Which application type would you like to use ?",
      "x-priority": "important"
    }
  },
  "required": ["directory", "type"]
}
