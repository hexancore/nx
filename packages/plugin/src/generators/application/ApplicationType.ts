export type ApplicationType = 'frontend' | 'backend';