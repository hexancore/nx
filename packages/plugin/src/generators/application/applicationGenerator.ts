import {
  Tree,
} from '@nx/devkit';
import * as path from 'path';
import { HcNxHelper } from '../../util/HcNxHelper';
import { ProjectMeta} from '../../util';
import { ApplicationGeneratorSchema } from './schema';

export async function applicationGenerator(tree: Tree, options: ApplicationGeneratorSchema): Promise<void> {

  if (options.type === 'frontend') {
    throw new Error('Not supported yet');
  }

  const helper = new HcNxHelper(tree);
  const project: ProjectMeta = helper.project.addApplication(options.directory, options.type);

  helper.project.generateFiles(project, helper.project.getCommonProjectFilesDir(options.type), '');
  helper.project.generateFiles(project, __dirname + `/files/${options.type}`, '');

  tree.write(path.join(project.root, 'test/helper/.gitkeep'), '');
  tree.write(path.join(project.root, 'test/integration/.gitkeep'), '');
}

export default applicationGenerator;
