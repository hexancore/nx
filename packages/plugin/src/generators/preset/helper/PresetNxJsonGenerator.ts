import {
  Tree,
  updateJson
} from '@nx/devkit';

export class PresetNxJsonGenerator {

  public static run(tree: Tree) {
    const g = new this();
    g.apply(tree);
  }

  public apply(tree: Tree): void {
    updateJson(tree, 'nx.json', (o) => {
      o.extends = 'nx/presets/npm.json';
      o.pluginsConfig = {
        "@nx/js": {
          "analyzeSourceFiles": true
        }
      };

      o.namedInputs = { ...(o.namedInputs ?? {}), ...this.namedInputs() };
      o.targetDefaults = { ...(o.targetDefaults ?? {}), ...this.targetDefaults() };
      o.generators = {
        "@nx/workspace:move": {
          "projectNameAndRootFormat": "as-provided"
        }
      };
      o.useInferencePlugins = false;
      o.defaultBase = 'main';

      o['workspaceLayout'] = {
        "appsDir": "apps",
        "libsDir": "libs"
      };

      return o;
    });
  }

  private namedInputs(): Record<string, any> {
    return {
      "default": [
        "{projectRoot}/**/*",
        "!{projectRoot}/README.md",
        "sharedGlobals"
      ],
      "sharedGlobals": [],
      production: [
        "default",
        "!{projectRoot}/test/**/*",
        "!{projectRoot}/.vite/**/*",
        "!{projectRoot}/.storybook/**/*",
        "!{projectRoot}/jest.config.{js,ts,mjs,mts}",
        "!{projectRoot}/tsconfig.test.json",
        "!{projectRoot}/.eslintrc.json"
      ]
    };
  }

  private targetDefaults(): Record<string, any> {
    return {
      "@nx/js:tsc": {
        "inputs": ["production", "^production"],
        "outputs": ["{options.outputPath}"],
        "dependsOn": ["^build", "test"],
        "cache": true,
        "options": {
          "outputPath": "dist/{projectRoot}",
          "tsConfig": "{projectRoot}/tsconfig.build.json"
        }
      },

      "@nx/jest:jest": {
        "cache": true,
        "inputs": ["default", "^production", "{workspaceRoot}/jest.preset.js"],
        "options": {
          "jestConfig": "{projectRoot}/jest.config.ts",
          "runInBand": true,
          "passWithNoTests": true
        },
        "configurations": {
          "watch": {
            "watch": true
          },
          "ci": {
            "ci": true,
            "codeCoverage": true
          },
          "updateSnapshot": {
            "updateSnapshot": true
          },
          "codeCoverage": {
            codeCoverage: true,
          }
        }
      },
      "@nx/eslint:lint": {
        "inputs": ["default", "{workspaceRoot}/.eslintrc.json", "{workspaceRoot}/.eslintignore"],
        "cache": true,
        "options": {
          "lintFilePatterns": ["{projectRoot}/**/*.ts", "{projectRoot}/package.json"]
        },
        "configurations": {
          "fix": {
            "fix": true,
            "cache": false
          }
        }
      },
      ...this.viteTargetsDefaults(),
      ...this.storybookTargetsDefaults(),
    };
  }

  private viteTargetsDefaults(): Record<string, any> {
    return {
      "@nx/vite:build": {
        "inputs": ["production", "^production"],
        "outputs": ["{options.outputPath}"],
        "defaultConfiguration": "production",
        "options": {
          "outputPath": "dist/{projectRoot}",
          "skipTypeCheck": true,
          "configFile": "{projectRoot}/vite.config.ts",
          "tsConfig": "{projectRoot}/tsconfig.build.json",
          "generatePackageJson": false
        },
        "configurations": {
          "development": {
            "mode": "development"
          },
          "production": {
            "mode": "production"
          }
        }
      },
      "@nx/vite:preview-server": {
        "defaultConfiguration": "development",
        "options": {
          "buildTarget": "build"
        },
        "configurations": {
          "development": {
            "buildTarget": "build:development"
          },
          "production": {
            "buildTarget": "build:production"
          }
        }
      },
      "@nx/vite:test": {
        "cache": true,
        "inputs": ["default", "^default"],
        "options": {
          "configFile": "{projectRoot}/vite.config.ts",
          "passWithNoTests": true,
          "reportsDirectory": "../../../coverage/{projectRoot}"
        },
        "configurations": {
          "watch": {
            "watch": true
          }
        }
      },
    };
  }

  private storybookTargetsDefaults(): Record<string, any> {
    return {
      "@nx/storybook:storybook": {
        "options": {
          "port": 4400,
          "configDir": "libs/front/mao1/.storybook"
        },
        "configurations": {
          "ci": {
            "quiet": true
          }
        }
      },
      "@nx/storybook:build": {
        "outputs": ["{options.outputDir}"],
        "options": {
          "outputDir": "dist/storybook/{projectRoot}",
          "configDir": "{projectRoot}/.storybook"
        },
        "configurations": {
          "ci": {
            "quiet": true
          }
        }
      },
      "storybook-test": {
        "executor": "nx:run-commands",
        "options": {
          "command": "test-storybook -c {projectRoot}/.storybook --url=http://localhost:4400"
        }
      },
      "storybook-static": {
        "executor": "@nx/web:file-server",
        "options": {
          "buildTarget": "storybook-build",
          "staticFilePath": "dist/storybook/{projectRoot}",
          "spa": true
        },
        "configurations": {
          "ci": {
            "buildTarget": "storybook-build:ci"
          }
        }
      }
    };
  }
}