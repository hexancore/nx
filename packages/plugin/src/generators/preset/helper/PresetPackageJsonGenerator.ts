import {
  addDependenciesToPackageJson,
  updateJson,
  Tree,
} from '@nx/devkit';

export class PresetPackageJsonGenerator {

  public constructor(
    private nxVersion: string,
    private hcNxPluginVersion: string
  ) {

  }

  public static run(tree: Tree, options: {
    nxVersion: string,
    hcNxPluginVersion: string
  }) {
    const g = new this(options.nxVersion, options.hcNxPluginVersion);
    g.apply(tree);
  }

  public apply(tree: Tree): void {
    this.generateDeps(tree);

    updateJson(tree, 'package.json', (o) => {
      o.license = 'UNLICENSED';
      o.scripts = o.scripts ?? {};

      o.scripts['prepare'] = "node .husky/install.mjs",
        o.scripts['precommit'] = "nx run-many --targets=lint,test,build";
      o.scripts['ci'] = "nx run-many --targets=lint,test,build,e2e --configuration=ci";
      return o;
    });
  }

  private generateDeps(tree: Tree): void {
    const deps = this.createDeps();
    const devDeps = this.createDevDeps();
    addDependenciesToPackageJson(tree, deps, devDeps);
  }

  private createDeps(): Record<string, string> {
    const hexancore = {
      '@hexancore/core': '0.14.*',
      '@hexancore/common': '0.13.*',
      '@hexancore/cloud': '0.1.*',
      '@hexancore/auth': '0.3.*',
      '@hexancore/typeorm': '0.14.*',
    };

    return {
      ...hexancore,
      'tslib': '2.6.2'
    };
  }

  private createDevDeps(): Record<string, string> {
    return {
      ...this.createHexancoreDevDeps(),
      ...this.createNxDevDeps(),
      ...this.createFrontendDevDeps(),
      ...this.createJestDevDeps(),
      ...this.createEslintDevDeps(),
      ...this.createEssentialsDevDeps()
    };
  }

  private createNxDevDeps(): Record<string, string> {
    return {
      "@nx/devkit": this.nxVersion,
      "@nx/esbuild": this.nxVersion,
      "@nx/eslint-plugin": this.nxVersion,
      "@nx/jest": this.nxVersion,
      "@nx/js": this.nxVersion,
      "@nx/linter": this.nxVersion,
      "@nx/plugin": this.nxVersion,
    };
  }

  private createHexancoreDevDeps(): Record<string, string> {
    const [major, minor] = this.hcNxPluginVersion.split('.');

    return {
      "@hexancore/nx": `${major}.${minor}.*`,
      "@hexancore/mocker": "^1.1.2",
    };
  }

  private createFrontendDevDeps(): Record<string, string> {
    const vue = {
      "@vue/eslint-config-typescript": "^11.0.3",
      "@vue/test-utils": "^2.4.5",
      "@vue/tsconfig": "^0.4.0",
      "vue": "^3.4.25",
      "vue-eslint-parser": "^9.4.2",
      "vue-router": "^4.3.2",
      "pinia": "^2.1.7",
      "vue-tsc": "^1.8.27",
      "vee-validate": "^4.12.6",
    };

    const vueI18n = {
      "vue-i18n": "^9.13.1",
      "@intlify/eslint-plugin-vue-i18n": "^2.0.0",
      "@intlify/unplugin-vue-i18n": "^2.0.0",
      "unplugin-swc": "^1.4.5",
    };

    const vite = {
      "@vitejs/plugin-basic-ssl": "^1.1.0",
      "@vitejs/plugin-vue": "^5.0.4",
      "@vitest/coverage-v8": "1.6.*",
      "@vitest/ui": "1.6.*",
      "vitest": "1.6.*",
      "@rollup/plugin-node-resolve": "^15.2.3",
      "rollup": "^4.17.0",
      "vite": "^5.2.10",
      "vite-tsconfig-paths": "^4.3.2",
      "autoprefixer": "^10.4.19",
      "@babel/core": "^7.24.4",
      "@babel/plugin-transform-modules-commonjs": "^7.24.1",
      "@babel/preset-env": "^7.24.4",
    };

    const primevue = {
      "primeflex": "^3.3.1",
      "primeicons": "^6.0.1",
      "primevue": "^3.52.0",
    };

    const style = {
      "postcss": "^8.4.38",
      "sass": "^1.75.0",
      "stylelint": "^15.11.0",
      "stylelint-config-standard": "^33.0.0",
      "@samatech/postcss-basics": "^0.6.0",
    };

    return {
      ...vue,
      ...vueI18n,
      ...vite,
      ...primevue,
      ...style
    };
  }

  public createEslintDevDeps(): Record<string, string> {
    return {
      "eslint": "8.57.0",
      "eslint-plugin-import": "^2.29.1",
      "eslint-plugin-vue": "^9.25.0",
      "@typescript-eslint/parser": "7.3.0",
      "@stylistic/eslint-plugin": "^2.1.0",
    };
  }

  public createJestDevDeps(): Record<string, string> {
    return {
      "happy-dom": "^9.20.3",
      "jest": "^29.7.0",
      "jest-docblock": "^29.7.0",
      "jest-environment-jsdom": "^29.7.0",
      "jest-environment-node": "^29.7.0",
      "jest-expect-message": "^1.1.3",
      "jest-runner": "^29.7.0",
      "jest-runner-groups": "^2.2.0",
      "jsdom": "~22.1.0",
      "ts-jest": "^29.1.2",
      "@types/jest": "^29.5.12",
      "@jest/globals": "^29.7.0",
    };
  }

  public createEssentialsDevDeps(): Record<string, string> {
    return {
      "tsconfig": "^7.0.0",
      "typescript": "5.4.5",
      "typescript-transform-paths": "^3.4.7",
      "esbuild": "^0.19.12",
      "axios-mock-adapter": "^1.22.0",
      "@types/node": "20.12.12",
      "@nestjs/testing": "^10.3.8",
      "husky": "^9.0.11",
    };
  }
}