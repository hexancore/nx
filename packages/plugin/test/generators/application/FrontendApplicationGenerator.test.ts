import { Tree } from '@nx/devkit';
import { applicationGenerator } from '../../../src/generators/application/applicationGenerator';
import type { ApplicationGeneratorSchema } from '../../../src/generators/application/schema';
import { createEmptyNxTree, exceptWorkspaceFileMatchSnapshot, expectedFiles } from '../../helper/functions';

describe('Application Generator: Frontend', () => {
  let tree: Tree;
  const type = 'frontend';
  const projectDirectory = `acme/${type}`;
  const projectRoot = `apps/${projectDirectory}`;
  beforeAll(async () => {
    tree = createEmptyNxTree();
    const options: ApplicationGeneratorSchema = { directory: projectDirectory, type };
    await applicationGenerator(tree, options);
  });

  const expectedProjectSrcFiles = expectedFiles(`${projectRoot}/src/`, [
    'constants.ts',
    'Core/Asset/Image/logo.png',
    'Core/Asset/Locale/en.yaml',
    'Core/Asset/Style/colors.module.css',
    'Core/Asset/Style/font.scss',
    'Core/Asset/Style/style.css',
    'Core/Asset/Style/theme.css',
    'Core/Asset/Style/theme.dark.css',
    'Core/Asset/Style/theme.light.css',
    'Core/Asset/Style/tokens.css',
    'Core/Asset/Style/typography.module.css',
    'Core/Asset/Style/variables.scss',
    'Core/Component/Layout/AppLayout.vue',
    'Core/Component/Misc/AppLogo.vue',
    'Core/Component/View/App.vue',
    'Core/Component/View/CoreRoutes.ts',
    'Core/Component/View/Dashboard.vue',
    'Core/Component/View/Error/404.vue',
    'Core/Service/Plugin/I18NCorePlugin.ts',
    'Core/Service/Plugin/PiniaCorePlugin.ts',
    'Core/Service/Plugin/PrimeVueCorePlugin.ts',
    'Core/Service/Router/Router.ts',
    'env.d.ts',
    'shims-vue.d.ts'
  ]);

  const expectedProjectFilesToMatchSnapshots = expectedFiles(`${projectRoot}/`, [
    'README.md',
    'project.json',
    '.eslintrc.json',
    'tsconfig.json',
    'tsconfig.test.json',
    'tsconfig.build.json',
    'tsconfig.storybook.json',
    'src/main.ts',
    '.storybook/main.ts',
    '.storybook/preview.ts',
    'vite.config.mts',
    '.env.development',
    'index.html'
  ]);

  const expectedWorkspaceFilesToMatchSnapshots = [
    ...expectedProjectFilesToMatchSnapshots,
    '.prettierrc',
    'package.json',
    'nx.json',
    'tsconfig.base.json',
  ];

  test.each(expectedWorkspaceFilesToMatchSnapshots.map(i => [i]))('%s should match snapshot', (filePath) => {
    exceptWorkspaceFileMatchSnapshot(tree, filePath);
  });

  test('only file from list should exist', () => {
    const expectedWorkspaceFilesToExists = [
      `${projectRoot}/test/helper/.gitkeep`,
      `${projectRoot}/test/unit/sample.test.ts`,
      `${projectRoot}/test/integration/.gitkeep`,
      ...expectedWorkspaceFilesToMatchSnapshots,
      ...expectedProjectSrcFiles,
      `${projectRoot}/public/favicon.ico`,
    ];
    const diff = tree.listChanges().map(c => c.path).filter(x => !expectedWorkspaceFilesToExists.includes(x));
    expect(diff).toEqual([]);
  });
});
