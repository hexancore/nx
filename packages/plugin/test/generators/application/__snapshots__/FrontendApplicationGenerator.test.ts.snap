// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Generator: Frontend .prettierrc should match snapshot 1`] = `"{"singleQuote":true}"`;

exports[`Application Generator: Frontend apps/acme/frontend/.env.development should match snapshot 1`] = `
"VITE_APP_API_URL=http://localhost:25021
"
`;

exports[`Application Generator: Frontend apps/acme/frontend/.eslintrc.json should match snapshot 1`] = `
"{
  "extends": [
    "plugin:vue/vue3-essential",
    "eslint:recommended",
    "@vue/eslint-config-typescript",
    "@vue/eslint-config-prettier/skip-formatting",
    "../../../.eslintrc.json"
  ],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx", "*.vue"],
      "rules": {
        "vue/multi-word-component-names": "off"
      }
    }
  ]
}
"
`;

exports[`Application Generator: Frontend apps/acme/frontend/.storybook/main.ts should match snapshot 1`] = `
"import {defineProjectStorybookConfig} from "../../../.workspace/storybook/main.preset";

export default defineProjectStorybookConfig('apps/acme/frontend');
"
`;

exports[`Application Generator: Frontend apps/acme/frontend/.storybook/preview.ts should match snapshot 1`] = `"export const tags = ["autodocs"];"`;

exports[`Application Generator: Frontend apps/acme/frontend/README.md should match snapshot 1`] = `
"# app-acme-frontend

"
`;

exports[`Application Generator: Frontend apps/acme/frontend/index.html should match snapshot 1`] = `
"<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <link rel="icon" href="/favicon.ico" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>app-pmt-web</title>
  </head>
  <body>
    <div id="root"></div>
    <script>
      window.process = window.process ?? { env: {} };
    </script>
    <script type="module" src="/src/main.ts"></script>
  </body>
</html>
"
`;

exports[`Application Generator: Frontend apps/acme/frontend/project.json should match snapshot 1`] = `
"{
  "name": "app-acme-frontend",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "projectType": "application",
  "sourceRoot": "{projectRoot}/src",
  "targets": {
    "lint": {
      "executor": "@nx/eslint:lint"
    },
    "test": {
      "executor": "@nx/vite:test"
    },
    "build": {
      "executor": "@nx/vite:build"
    },
    "preview": {
      "executor": "@nx/vite:preview-server"
    },
    "storybook": {
      "executor": "@nx/storybook:storybook"
    },
    "storybook-test": {
      "executor": "nx:run-commands"
    },
    "storybook-build": {
      "executor": "@nx/storybook:build"
    },
    "storybook-static": {
      "executor": "@nx/web:file-server"
    }
  }
}
"
`;

exports[`Application Generator: Frontend apps/acme/frontend/src/main.ts should match snapshot 1`] = `
"process = process ?? { env: {} };

import { createApp } from 'vue';
import App from './Core/Component/View/App.vue';

import { PrimeVueCorePlugin } from './Core/Service/Plugin/PrimeVueCorePlugin';
import router from './Core/Service/Router/Router';
import { I18NCorePlugin } from './Core/Service/Plugin/I18NCorePlugin';
import { PiniaCorePlugin } from './Core/Service/Plugin/PiniaCorePlugin';

const app = createApp(App);

PrimeVueCorePlugin(app);
I18NCorePlugin(app);
PiniaCorePlugin(app);
app.use(router);

app.mount('#root');
"
`;

exports[`Application Generator: Frontend apps/acme/frontend/tsconfig.build.json should match snapshot 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../../dist/out-tsc",
    "types": ["vite/client"]
  },
  "exclude": [
    "**/*.stories.ts"
  ],
  "include": ["src/**/*.ts","src/**/*.tsx", "src/**/*.vue"]
}
"
`;

exports[`Application Generator: Frontend apps/acme/frontend/tsconfig.json should match snapshot 1`] = `
"{
  "extends": "../../../tsconfig.base.json",
  "compilerOptions": {
    "allowJs": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "jsx": "preserve",
    "jsxImportSource": "vue",
    "resolveJsonModule": true,
    "verbatimModuleSyntax": false,
    "useDefineForClassFields": true,
    "isolatedModules": true,
    "types": ["vite/client", "vitest", "@intlify/unplugin-vue-i18n/messages"],
    "noEmit": true,
    "lib": ["ES2022", "DOM"],
    "skipLibCheck": true,

    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "paths": {
      "@": [
        "apps/acme/frontend/src"
      ],
      "@/*": [
        "apps/acme/frontend/src/*"
      ],
      "@test/*": [
        "apps/acme/frontend/test/helper/*"
      ]
    }
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.build.json"
    },
    {
      "path": "./tsconfig.test.json"
    },
    {
      "path": "./tsconfig.storybook.json"
    }
  ]
}
"
`;

exports[`Application Generator: Frontend apps/acme/frontend/tsconfig.storybook.json should match snapshot 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "emitDecoratorMetadata": true
  },
  "include": [
    "src/**/*.stories.ts",
    "src/**/*.stories.tsx",
    ".storybook/*.ts"
  ]
}
"
`;

exports[`Application Generator: Frontend apps/acme/frontend/tsconfig.test.json should match snapshot 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../../dist/out-tsc",
    "types": ["vitest/globals", "vitest/importMeta", "vite/client", "node", "vitest"]
  },
  "include": ["vite.config.ts", "test/**/*.ts", "test/**/*.tsx"]
}
"
`;

exports[`Application Generator: Frontend apps/acme/frontend/vite.config.ts should match snapshot 1`] = `
"import { defineAppProjectViteConfig } from "../../../.workspace/vite/vite.preset";

const project = {
  workspaceRootRelative: '../../..',
  root: 'apps/acme/frontend',
  name: 'app-acme-frontend'
};

export default defineAppProjectViteConfig(__dirname, project);
"
`;

exports[`Application Generator: Frontend nx.json should match snapshot 1`] = `"{"affected":{"defaultBase":"main"},"targetDefaults":{"build":{"cache":true},"lint":{"cache":true}}}"`;

exports[`Application Generator: Frontend package.json should match snapshot 1`] = `
"{
  "name": "@testcompany/source"
}
"
`;

exports[`Application Generator: Frontend tsconfig.base.json should match snapshot 1`] = `"{"compilerOptions":{"paths":{}}}"`;
